problem_builder/plot.py

```diff
@@ -346,6 +346,27 @@ def student_view(self, context=None):
         fragment.initialize_js('PlotBlock')
         return fragment
 
+    def student_view_data(self, context=None):
+        """
+        Returns a JSON representation of the student_view of this XBlock,
+        retrievable from the Course XBlock API.
+        """
+        return {
```

Comment:
This block does not support `self.name` (only blocks that include the `QuestionMixin` do). I was wrong to suggest adding the `id` field here.